src/main/java/com/algosenpai/app/chapters/ChapterSorting.java

```diff
@@ -0,0 +1,293 @@
+package com.algosenpai.app.chapters;
+
+import com.algosenpai.app.Question;
+import com.algosenpai.app.model.ReviewTracingListModel;
+import com.algosenpai.app.model.ReviewTracingModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.Scanner;
+import java.util.HashSet;
+
+public class ChapterSorting {
+
+    private static Random random = new Random();
+    private static Scanner s = new Scanner(System.in);
+    /**
+     * Generates the question by using a random number to determine which of the
+     * sub-questions to ask.
+     * 
+     * @return A question class that contains the question and expected answer.
+     */
+
+    public static Question generateQuestions() {
+        int questionType = random.nextInt(4);
+        switch (questionType) {
+        case 0:
+            return bubbleSortPassesQuestion();
+        case 1:
+            return quickSortPivotsQuestion();
+        case 2:
+            return insertionSortSwapsQuestion();
+        case 3:
+            return selectionSortSwapsQuestion();
+        default:
+            return null;
+        }
+    }
+
+    /**
+     * Generates the Selection Sort Algorithm question related to the sequence of
+     * integers after a certain number of swaps. It determines a random array size,
+     * then runs the algorithm on it for certain number of swaps, after which it
+     * will terminate.
+     * 
+     * @return the question class with the question.
+     */
+    private static Question selectionSortSwapsQuestion() {
+        int arraySize = random.nextInt(4) + 5;
+        ArrayList<Integer> initialArray = new ArrayList<>(generateArray(arraySize));
+        int swaps = random.nextInt(arraySize - 5) + 1;
+        String question = "An array of " + arraySize + " elements underwent the following Selection Sort Algorithm : "
+                + initialArray + "\n";
+        question += "What would be the new configuration of the elements after " + swaps + " swaps?\n";
+        question += "int i, j, min_idx;\n" + "   for (i = 0; i < n-1; i++) {\n" + "       min_idx = i;\n"
+                + "       for (j = i+1; j < n; j++) {\n" + "           if (arr[j] < arr[min_idx]) {\n"
+                + "               min_idx = j;\n" + "       }\n" + "       }\n" + "   if (min_idx != i) {"
+                + "       swap(arr[min_idx], arr[i]);\n" + "   }" + "}\n";
```

Comment:
Still equivalent to magic numbers here. You should make these chunk of code to be a (final) variable, and do tidy up the code so that readers can understand.